airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/pipeline.py

```diff
@@ -431,9 +458,35 @@ async def _run(self) -> StepResult:
         return await super()._run()
 
 
+class CheckConnectorVersionIncrement(Step):
+    context: PublishConnectorContext
+    
+    @property
+    def title(self) -> str:
+        return "Check if connector version was incremented"
+
+    async def _run(self) -> StepResult:
+        # Skip if running from manual GitHub workflow
+        if os.getenv("GITHUB_WORKFLOW_MANUAL") == "true":
+            return StepResult(step=self, status=StepStatus.SKIPPED, stdout="Skipping version increment check for manual workflow run.")
+
+        # Skip if connector opts out of version checks
+        if self.context.metadata and self.context.metadata.get("ab_internal", {}).get("requireVersionIncrementsInPullRequests") is False:
```

Comment:
You should not perform this opt-out here. IF the version was not incremented, publish should NOT happen, so we should not skip this step.